maven-jahia-plugin/src/main/java/org/jahia/utils/maven/plugin/osgi/SourceFolderMojo.java

```diff
@@ -0,0 +1,174 @@
+import org.apache.maven.artifact.handler.manager.ArtifactHandlerManager;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.project.MavenProjectHelper;
+import org.apache.tika.io.IOUtils;
+import org.codehaus.plexus.archiver.UnArchiver;
+import org.codehaus.plexus.archiver.jar.JarArchiver;
+import org.codehaus.plexus.archiver.manager.ArchiverManager;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+/**
+ * A goal that checks the dependencies of a generated OSGi bundle JAR against the project dependencies, and reports
+ * any missing packages that weren't found in any dependency export.
+ *
+ * @goal update-jahia-source-folder
+ * @phase package
+ */
+public class SourceFolderMojo extends AbstractMojo {
+    /**
+     * @parameter expression="${project}"
+     * @readonly
+     * @required
+     */
+    private MavenProject project;
+
+    /**
+     * Classifier type of the bundle to be installed.  For example, "jdk14".
+     * Defaults to none which means this is the project's main bundle.
+     *
+     * @parameter
+     */
+    protected String classifier;
+
+    /**
+     * @component
+     */
+    private ArchiverManager archiverManager;
+
+    /**
+     * @component
+     */
+    private ArtifactHandlerManager artifactHandlerManager;
+
+    /**
+     * @component
+     */
+    private MavenProjectHelper mavenProjectHelper;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        String extension = project.getPackaging();
+        if ("bundle".equals(extension)) {
+            extension = "jar";
+        }
+
+        String artifactFilePath = project.getBuild().getDirectory() + File.pathSeparator + project.getBuild().getFinalName() + "." + extension;
+
+        File artifactFile = new File(artifactFilePath);
+        if (!artifactFile.exists()) {
+            throw new MojoExecutionException("No artifact generated for project, was the goal called in the proper phase (should be verify) ?");
+        }
+        try (JarFile jarFile = new JarFile(artifactFile)) {
+            Manifest manifest = jarFile.getManifest();
+            if (manifest.getMainAttributes() == null) {
+                throw new MojoExecutionException("Error reading OSGi bundle manifest data from artifact " + artifactFile);
+            }
+            String groupId = project.getGroupId();
+            boolean isModule = groupId.equals("org.jahia.module") || groupId.endsWith(".jahia.modules") || manifest.getMainAttributes().containsKey(new Attributes.Name("Jahia-Module-Type"));
+            if (isModule) {
+                String sourceFolder = "/data/digital-factory-data/sources/" + project.getBasedir().getName();
```

Comment:
shouldn't we use /var/jahia by default ?

Reply:
good catch, forgot this one

Reply:
in fact it is not activated and as we now try to find the sources in the data folder of Jahia during runtime if they are not found in Jahia-Source-Folders entry, we should probably get rid of this class

Reply:
@tdraier

Reply:
yes i was not clear about the usage of it, you can remove it